### PR TITLE
Add grant for canSubscribeMetrics.

### PR DESCRIPTION
**Pull request by boks1971**

### CODE REVIEW

File: auth/grants.go [hunk 1]
```diff
@@ -78,6 +78,8 @@ type ClaimGrants struct {
 	Metadata string `json:"metadata,omitempty"`
 	// Key/value attributes to attach to the participant
 	Attributes map[string]string `json:"attributes,omitempty"`
+	// if a participant can subscribe to metrics
+	CanSubscribeMetrics *bool `json:"canSubscribeMetrics,omitempty"`
```
Comment (davidzhao) [inline auth/grants.go:82]: can we put this into `Video` grant? this is where the other permission bits are stored. not named the best, but seems more consistent given where things are today.

Comment (boks1971) [reply auth/grants.go:82]: okay, I put it here because of the naming.

Comment (boks1971) [reply auth/grants.go:98]: Moved it in this [commit](https://github.com/livekit/protocol/pull/795/commits/1add410bf995f5e8e69926accf3e517fd187f908). A few more things
- Grouped struct with its methods (makes it easier on my brain :-) )
- It defaults to `false`, so not adding to things like `BuildSIPToken`, `BuildAgentToken`.

**Review by davidzhao — APPROVED** [review]
lgtm

**Pull request merged**